### PR TITLE
Update to React 18

### PR DESCRIPTION
**Pull request by amoore108**
**Overall change:**
- Updates Simorgh to React 18. Doing this also allows the NextJS client to use Next v13 https://nextjs.org/blog/next-13 
- Replaces old `hydrate` function with new `hydrateRoot` function
- Leaves `renderToString` in place for now. The newer API `renderToPipeableStream` is recommended to replace this https://reactjs.org/docs/react-dom-server.html#rendertopipeablestream, however not necessary as we don't make use of Suspense yet
- Updates testing libraries to latest versions to be compatible with React 18

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/client.test.jsx [hunk 1]
```diff
@@ -47,27 +47,27 @@ describe('Client', () => {
     resetWindowValue('location', windowLocation);
   });
 
-  it('should hydrate client once routes are ready', async () => {
+  it('should hydrateRoot client once routes are ready', async () => {
```
Comment (karinathomasbbc) [inline src/client.test.jsx]: Find and replace error? 😸 
```suggestion
  it('should hydrate client once routes are ready', async () => {
```

File: src/client.test.jsx [hunk 2]
```diff
@@ -47,27 +47,27 @@ describe('Client', () => {
...
 
-        expect(reactDom.hydrate).toHaveBeenCalled();
+        expect(reactDom.hydrateRoot).toHaveBeenCalled();
         resolve();
       });
     });
   });
 
-  it('should not hydrate client if no routes match', async () => {
+  it('should not hydrateRoot client if no routes match', async () => {
```
Comment (karinathomasbbc) [inline src/client.test.jsx]:
```suggestion
  it('should not hydrate client if no routes match', async () => {
```

Comment (amoore108) [reply src/client.test.jsx]: 😱

**Review by karinathomasbbc — APPROVED** [review]
Could we get this onto a test environment and run E2Es against it?

**Comment by amoore108** [comment]
> Could we get this onto a test environment and run E2Es against it?

Yea definitely. Would like to check Opera Mini on this as well when its on Test.

**Pull request merged**